packages/hash/frontend/src/pages/[shortname]/entities/[entity-uuid].page/create-entity-page.tsx

```diff
@@ -0,0 +1,169 @@
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- @todo improve logic or types to remove this comment
+    (activeWorkspaceAccountId as OwnedById) ?? null,
+  );
+
+  const { getEntityType } = useBlockProtocolGetEntityType();
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        setLoading(true);
+
+        const { data: subgraph } = await getEntityType({
+          data: {
+            entityTypeId,
+            graphResolveDepths: {
+              constrainsValuesOn: { outgoing: 255 },
+              constrainsLinksOn: { outgoing: 255 },
+              constrainsLinkDestinationsOn: { outgoing: 255 },
+              constrainsPropertiesOn: { outgoing: 255 },
+            },
+          },
+        });
+
+        if (!subgraph) {
+          throw new Error("subgraph not found");
+        }
+
+        const draftEntityEditionId: EntityEditionId = {
+          baseId: "draft%draft",
+          version: new Date().toISOString(),
+        };
+
+        setDraftEntitySubgraph({
+          ...subgraph,
+          roots: [draftEntityEditionId],
+          vertices: {
+            ...subgraph.vertices,
+            [draftEntityEditionId.baseId]: {
+              [draftEntityEditionId.version]: {
+                kind: "entity",
+                inner: {
+                  properties: {},
+                  metadata: {
+                    editionId: draftEntityEditionId,
+                    entityTypeId,
+                    provenance: { updatedById: "" },
+                    archived: false,
+                  },
+                },
+              },
+            },
+          },
+        } as Subgraph<SubgraphRootTypes["entity"]>);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    void init();
+  }, [entityTypeId, getEntityType]);
+
+  const [handleCreateEntity, creating] = useLoadingCallback(async () => {
+    if (!draftEntitySubgraph || !activeWorkspace) {
+      return;
+    }
+
+    const draftEntity = getRoots(draftEntitySubgraph)[0];
+
+    if (!draftEntity) {
+      return;
+    }
+
+    const { data: entity } = await createEntity({
+      data: {
+        entityTypeId,
+        properties: draftEntity.properties,
+      },
+    });
+
+    if (!entity) {
+      return;
+    }
+
+    const entityId = extractEntityUuidFromEntityId(
+      entity.metadata.editionId.baseId,
+    );
+
+    void router.push(`/@${activeWorkspace.shortname}/entities/${entityId}`);
+  });
+
+  if (loading) {
+    return <EntityPageLoadingState />;
+  }
+
+  if (!draftEntitySubgraph) {
+    return <PageErrorState />;
+  }
+
+  const entityLabel = generateEntityLabel(draftEntitySubgraph);
+
+  return (
+    <EntityEditorPage
+      /**
+       * @todo links section is hidden temporarily on new entity page
+       * it should be visible again after draft state on links implemented
+       * */
+      hideLinksSection
+      refetch={async () => {}}
```

Comment:
What about having a mode where you indicate you're creating, and let the component handle this instead – so you don't need to pass a refetch prop when mode = "creating", and it knows to hide the links section in that mode. Seems like a clearer API.

Reply:
We're going to get rid of the `refetch` soon, when we implement draft state for links as well, so I thought we don't want to modify the API for now.